src/seedu/addressbook/commands/Command.java

```diff
@@ -38,9 +38,16 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
 
     /**
      * Executes the command and returns the result.
+     * @throws Exception 
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws Exception;
 
+    
+    /**
+     * Returns true if the command mutates the data
```

Comment:
Try to be more specific on this header comment; the abstract class is a contract that we are providing to the client classes, hence the documentation should be precise. e.g., think what happens when you try to add a duplicate entry to address book. does it still mutate data?